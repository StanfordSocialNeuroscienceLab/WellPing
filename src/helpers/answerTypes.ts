import { AnswerEntity } from "../entities/AnswerEntity";
import { Question, QuestionsList, Choice } from "./types";

// We use `{ value: ... }` because it seems that `"simple-json"`
// does not support simply mixing normal values.
// This also increases the consistency of the data.

export type SliderAnswerData = {
  value: number;
};

export type ChoicesWithSingleAnswerAnswerData = {
  value: Choice;
};

// We use an array of tuples here (instead of `{ [key: string]: boolean }`)
// because the order shown on the screen can be kept in the results.
// This is potentially helpful for analysis later if the order is randomized
// as now we know the order the user saw the choices in.
export type ChoicesWithMultipleAnswersAnswerChoices = [Choice, boolean][];
export type ChoicesWithMultipleAnswersAnswerData = {
  value: ChoicesWithMultipleAnswersAnswerChoices;
};

export type YesNoAnswerData = {
  value: boolean;
};

export type MultipleTextAnswerData = {
  value: string[];
};

export type HowLongAgoAnswerDataType = [number | null, string | null];
export type HowLongAgoAnswerData = {
  value: HowLongAgoAnswerDataType;
};

export type AnswerData =
  | SliderAnswerData
  | ChoicesWithSingleAnswerAnswerData
  | ChoicesWithMultipleAnswersAnswerData
  | YesNoAnswerData
  | MultipleTextAnswerData
  | HowLongAgoAnswerData;

export interface AnswersList {
  [id: string]: AnswerEntity;
}

export interface QuestionScreenProps {
  question: Question;
  onDataChange: (data: AnswerData) => void;
  allAnswers: AnswersList;
  allQuestions: QuestionsList;
  pipeInExtraMetaData: (input: string) => string;
  setDataValidationFunction: (func: () => boolean) => void;
}
