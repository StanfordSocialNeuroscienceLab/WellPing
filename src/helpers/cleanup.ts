import { secureRemoveAllAnswersAsync } from "./secureStore/answer";
import { secureRemoveAllPingsAsync } from "./secureStore/ping";

export async function clearAllPingsAndAnswersAsync(): Promise<void> {
  await secureRemoveAllAnswersAsync();
  await secureRemoveAllPingsAsync();
}
