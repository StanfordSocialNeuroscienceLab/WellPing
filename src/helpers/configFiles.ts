import { isThisWeek } from "date-fns";

import { SurveyFile, Names, StudyID, StudyInfo, StreamName } from "./types";

export function getSurveyFile(): SurveyFile {
  const survey: SurveyFile = require("../../config/survey.json");
  return survey;
}

// TODO: DECOUPLE FUNCTIONS LIKE THIS
export function getNamesFile(): Names {
  const names: Names = require("../../config/names.json");
  return names;
}

export function getStudyInfo(): StudyInfo {
  return getSurveyFile().studyInfo;
}

export function getAllStreamNames(): StreamName[] {
  return Object.keys(getSurveyFile().meta.startingQuestionIds) as StreamName[];
}

export function getStudyId(): StudyID {
  return getStudyInfo().id;
}

export function isTimeThisWeek(time: Date): boolean {
  return isThisWeek(time, { weekStartsOn: getStudyInfo().weekStartsOn });
}
