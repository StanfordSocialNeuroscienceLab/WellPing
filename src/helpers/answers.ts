import { AnswerData, Answer } from "./answerTypes";
import { getAnswersPingIdsQuestionIdsListAsync } from "./asyncStorage/answersPingIdsQuestionIdsList";
import { AnswerSchema } from "./schemas/Answer";
import {
  secureGetAnswerAsync,
  secureStoreAnswerAsync,
} from "./secureStore/answer";
import { Question, Ping } from "./types";

export async function getAnswersAsync(): Promise<Answer[]> {
  const answersList = await getAnswersPingIdsQuestionIdsListAsync();

  // https://stackoverflow.com/q/28066429/2603230
  const answers: Answer[] = await Promise.all(
    answersList.map(async (pingIdAndQuestionId) => {
      const answer = await secureGetAnswerAsync(
        pingIdAndQuestionId[0],
        pingIdAndQuestionId[1],
      );
      if (answer === null) {
        throw new Error(
          "answer === null in answersList.map in getAnswersAsync.",
        );
      }
      return answer;
    }),
  );

  return answers;
}

export async function insertAnswerAsync({
  ping,
  question,
  realQuestionId,
  preferNotToAnswer,
  data,
  date,
}: {
  ping: Ping;
  question: Question;
  realQuestionId: string;
  preferNotToAnswer: true | null; // See `MARK: WHY_PNA_TRUE_OR_NULL`.
  data: AnswerData | null;
  date: Date;
}): Promise<Answer> {
  const answer = AnswerSchema.parse({
    pingId: ping.id,
    questionId: realQuestionId,
    preferNotToAnswer,
    data,
    date,
  });

  await secureStoreAnswerAsync(answer);

  return answer;
}
