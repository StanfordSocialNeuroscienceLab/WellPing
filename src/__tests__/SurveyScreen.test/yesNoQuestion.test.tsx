import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  RenderAPI,
} from "react-native-testing-library";

import SurveyScreen, { SurveyScreenProps } from "../../SurveyScreen";
import { QuestionType } from "../../helpers/helpers";
import {
  getBaseProps,
  setUpSurveyScreenTestAsync,
  tearDownSurveyScreenTestAsync,
  testQuestionsSequenceAsync,
} from "./helper";

let currentPropsBase!: SurveyScreenProps;
beforeEach(async () => {
  const currentTestPing = await setUpSurveyScreenTestAsync();
  currentPropsBase = {
    ...getBaseProps(),
    questions: {},
    startingQuestionId: "q1",
    ping: currentTestPing,
  };
});

afterEach(async () => {
  await tearDownSurveyScreenTestAsync();
});

async function clickOptionAsync(
  yesNo: "Yes" | "No",
  { findByA11yLabel }: RenderAPI,
) {
  // `findBy` does the `waitFor` for us.
  fireEvent.press(await findByA11yLabel(`select ${yesNo}`));
}

describe("with both branchStartId", () => {
  // See `MARK: SURVEY_TEST_WHY_GET_PROPS`.
  const getProps = (): SurveyScreenProps => ({
    ...currentPropsBase,
    questions: {
      q1: {
        id: "q1",
        type: QuestionType.YesNo,
        question: "Question 1",
        branchStartId: {
          yes: "q1.yes",
          no: "q1.no",
        },
        next: "q2",
      },
      "q1.yes": {
        id: "q1.yes",
        type: QuestionType.YesNo,
        question: "Question 1 - yes branch",
        next: null,
      },
      "q1.no": {
        id: "q1.no",
        type: QuestionType.YesNo,
        question: "Question 1 - no branch",
        next: null,
      },
      q2: {
        id: "q2",
        type: QuestionType.Slider,
        question: "Question 2",
        slider: ["left", "right"],
        next: null,
      },
    },
    startingQuestionId: "q1",
  });

  test("click yes", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1 - yes branch");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click no", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1 - no branch");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click next without answering", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click prefer not to answer", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            // Clicking "Yes" shouldn't matter here if we click prefer not
            // to answer.
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
          nextButton: "pna",
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });
});

describe("with only `yes` branchStartId", () => {
  const getProps = (): SurveyScreenProps => ({
    ...currentPropsBase,
    questions: {
      q1: {
        id: "q1",
        type: QuestionType.YesNo,
        question: "Question 1",
        branchStartId: {
          yes: "q1.yes",
        },
        next: "q2",
      },
      "q1.yes": {
        id: "q1.yes",
        type: QuestionType.YesNo,
        question: "Question 1 - yes branch",
        next: null,
      },
      q2: {
        id: "q2",
        type: QuestionType.Slider,
        question: "Question 2",
        slider: ["left", "right"],
        next: null,
      },
    },
    startingQuestionId: "q1",
  });

  test("click yes", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1 - yes branch");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click no", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click next without answering", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click prefer not to answer", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            // Clicking "Yes" shouldn't matter here if we click prefer not
            // to answer.
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
          nextButton: "pna",
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });
});

describe("with only `no` branchStartId", () => {
  const getProps = (): SurveyScreenProps => ({
    ...currentPropsBase,
    questions: {
      q1: {
        id: "q1",
        type: QuestionType.YesNo,
        question: "Question 1",
        branchStartId: {
          no: "q1.no",
        },
        next: "q2",
      },
      "q1.no": {
        id: "q1.no",
        type: QuestionType.YesNo,
        question: "Question 1 - no branch",
        next: null,
      },
      q2: {
        id: "q2",
        type: QuestionType.Slider,
        question: "Question 2",
        slider: ["left", "right"],
        next: null,
      },
    },
    startingQuestionId: "q1",
  });

  test("click yes", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click no", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1 - no branch");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click next without answering", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click prefer not to answer", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            // Clicking "No" shouldn't matter here if we click prefer not
            // to answer.
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
          nextButton: "pna",
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });
});

describe("with only `yes` and `no` being `null` branchStartId", () => {
  const getProps = (): SurveyScreenProps => ({
    ...currentPropsBase,
    questions: {
      q1: {
        id: "q1",
        type: QuestionType.YesNo,
        question: "Question 1",
        branchStartId: {
          yes: null,
          no: null,
        },
        next: "q2",
      },
      q2: {
        id: "q2",
        type: QuestionType.Slider,
        question: "Question 2",
        slider: ["left", "right"],
        next: null,
      },
    },
    startingQuestionId: "q1",
  });

  test("click yes", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
      ],
    });
  });

  test("click no", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
      ],
    });
  });

  test("click next without answering", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click prefer not to answer", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            // Clicking "No" shouldn't matter here if we click prefer not
            // to answer.
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
          nextButton: "pna",
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });
});

describe("with undefined branchStartId", () => {
  const getProps = (): SurveyScreenProps => ({
    ...currentPropsBase,
    questions: {
      q1: {
        id: "q1",
        type: QuestionType.YesNo,
        question: "Question 1",
        next: "q2",
      },
      q2: {
        id: "q2",
        type: QuestionType.Slider,
        question: "Question 2",
        slider: ["left", "right"],
        next: null,
      },
    },
    startingQuestionId: "q1",
  });

  test("click yes", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("Yes", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click no", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            await clickOptionAsync("No", renderResults);

            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click next without answering", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });

  test("click prefer not to answer", async () => {
    const onFinishFn = jest.fn();

    const renderResults = render(
      <SurveyScreen {...getProps()} onFinish={onFinishFn} />,
    );

    await testQuestionsSequenceAsync({
      renderResults,
      onFinishFn,
      sequence: [
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 1");
          },
          nextButton: "pna",
        },
        {
          expectCurrentQuestionAsync: async (getCurrentQuestionTitle) => {
            expect(getCurrentQuestionTitle()).toBe("Question 2");
          },
        },
      ],
    });
  });
});
