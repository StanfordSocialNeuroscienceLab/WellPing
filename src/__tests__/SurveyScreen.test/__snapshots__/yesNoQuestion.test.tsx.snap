// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with both branchStartId click next without answering: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with both branchStartId click no: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": false,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1_no",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with both branchStartId click prefer not to answer: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": true,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with both branchStartId click yes: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": true,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1_yes",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`no\` branchStartId click next without answering: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`no\` branchStartId click no: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": false,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1_no",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`no\` branchStartId click prefer not to answer: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": true,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`no\` branchStartId click yes: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": true,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` and \`no\` being \`null\` branchStartId click next without answering: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` and \`no\` being \`null\` branchStartId click no: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": false,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
]
`;

exports[`with only \`yes\` and \`no\` being \`null\` branchStartId click prefer not to answer: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": true,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` and \`no\` being \`null\` branchStartId click yes: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": true,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
]
`;

exports[`with only \`yes\` branchStartId click next without answering: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` branchStartId click no: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": false,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` branchStartId click prefer not to answer: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": true,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with only \`yes\` branchStartId click yes: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": true,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1_yes",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with undefined branchStartId click next without answering: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with undefined branchStartId click no: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": false,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with undefined branchStartId click prefer not to answer: getAnswersAsync 1`] = `
Array [
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": true,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;

exports[`with undefined branchStartId click yes: getAnswersAsync 1`] = `
Array [
  Object {
    "data": Object {
      "value": true,
    },
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q1",
  },
  Object {
    "data": null,
    "date": 1970-01-01T00:00:00.000Z,
    "pingId": "[removed for snapshot]",
    "preferNotToAnswer": null,
    "questionId": "q2",
  },
]
`;
